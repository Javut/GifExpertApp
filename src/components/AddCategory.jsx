
import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChange = (event) => {
    setInputValue(event.target.value);
    console.log(event.target.value);
  }

  const onSubmit = (event) => {
    console.log('Hola Mundo desde onSubmit');
    event.preventDefault();
    
    if(inputValue.trim().length <= 1) return;

    //setCategories(categories => [inputValue, ...categories]);
    onNewCategory(inputValue.trim().toUpperCase());
    setInputValue('');
    
    // setInputValue([...setCategories, inputValue]);
  }


  return (

    <form onSubmit={onSubmit} aria-label="form">
        <input
            type="text"
            placeholder='Buscar Gifs'
            value={inputValue}
            onChange={(event) => onInputChange(event)}
        />
    </form>

  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}
