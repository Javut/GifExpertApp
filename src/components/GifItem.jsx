
import React, { useState } from 'react'

export const GifItem = ({title, url}) => {

  const [first, setFirst] = useState();

  return (
    <div className="card">
      <img src={url} alt={title} />  
      <p>{title}</p>
    </div>
  )
}
