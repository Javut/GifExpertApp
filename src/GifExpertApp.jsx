
import React, { useState } from 'react'
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['One Punch']);


  const onAddCategory = (newCategory) => {


    if(categories.includes(newCategory)) return;

  //  if(categ.toLowerCase.includes(newCategory)) return;
    setCategories([newCategory]);
  }


  return (
    
    <>
      <h1>GifExpertApp</h1>

      <AddCategory 
        //setCategories={setCategories} -- Esta es una forma, enviando la funcion al componente hijo 
        onNewCategory={value => onAddCategory(value)}
        />

      <button onClick={onAddCategory}>Agregar</button>

      <ol>
      {
        categories.map((cat) => 
        <GifGrid 
          key={cat} 
          category={cat}
         />
        )
      }
      </ol>


    </>


)
}



  // DTHQiNDMMoza8g9suGB88BgqOCotRIoq