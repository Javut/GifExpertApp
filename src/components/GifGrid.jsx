import { useState } from "react";
import { useEffect } from "react";
import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    console.log(isLoading);


    // const [images, setImages] = useState([]);

    // const getImages = async () => {
    //     const newImages = await getGifs(category);
    //     setImages(newImages);
    // }

    // useEffect(() => {
    //     getImages();
    // }, [])


    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && (<h2>Cargando...</h2>)
                // isLoading ? ( <h2>Cargando...</h2>) : null
            }
           

            <div className="card-grid">
                {
                    images.map((img) => (
                        //    <li key={id}>{title}</li>
                        <GifItem
                           key={images.id}
                           title={img.title}
                           url={img.url}
                        //    {...img} Esta es otra forma de pasar las propiedades, esparciendolas al componnte Hijo
                        />
                    )
                    )
                }
            </div>

            {/* <h5>{counter}</h5> */}
            {/* <button onClick={() => setCounter(counter+1)}>+1</button>
        <p>Hola Mundo</p> */}
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
