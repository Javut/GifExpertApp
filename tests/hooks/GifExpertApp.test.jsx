import { fireEvent, render, screen } from "@testing-library/react"
import { AddCategory } from "../../src/components";
import { GifExpertApp } from "../../src/GifExpertApp"

describe('Pruebas en el componente <GifExpertApp />', () => {

    const title = 'GifExpertApp';
    const onNewCategory = jest.fn();

    test('Debe renderizar el title de la pagina ', () => {

        render(<GifExpertApp/>);

        const headingText = screen.getByText('GifExpertApp');

        expect(headingText).toMatchSnapshot();
      
    })


    test('Debe renderizar los componentes de la pagina ', () => {

        render(<GifExpertApp/>);
        const button = screen.getByRole('button')

        const headingText = screen.getByText('GifExpertApp');

        expect(headingText).toMatchSnapshot();
        // expect(<AddCategory/>).toBeTruthy();
        expect(button).toBeTruthy();
      
    })

    const newCategory = "Saitama";

    test('should Add new categories', () => {

        render(<GifExpertApp/>);
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, {target: {value: newCategory } });
        fireEvent.submit(form);

        fireEvent.input(input, {target: {value: newCategory + '2' } });
        fireEvent.submit(form);

        expect(screen.getAllByRole('heading', {level: 3}).length).toBe(3);

      
    })

    test('should no Add repeat category', () => {

        render(<GifExpertApp/>);
        screen.debug();
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, {target: {value: newCategory } });
        fireEvent.submit(form);

        fireEvent.input(input, {target: {value: newCategory } });
        fireEvent.submit(form);

        expect(screen.getAllByRole('heading', {level: 3}).length).toBe(2);


      
    })
    
    
  
})
