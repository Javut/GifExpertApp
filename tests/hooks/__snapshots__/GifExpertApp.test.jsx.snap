// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe renderizar el title de la pagina  1`] = `
<h1>
  GifExpertApp
</h1>
`;

exports[`Pruebas en el componente <GifExpertApp /> Debe renderizar los componentes de la pagina  1`] = `
<h1>
  GifExpertApp
</h1>
`;
